### PR TITLE
IA sync bot

### PR DESCRIPTION
**Pull request by hornc**
This code helps resolve and is referenced by https://github.com/internetarchive/openlibrary/issues/1046

## Description:
In this Pull Request we have made the following changes:
* Various scripts to check and sync archive.org and Open Library identifiers between the two systems

This is somewhat old code, but it was useful to sync up a large gap in identifiers, see https://github.com/internetarchive/openlibrary/issues/1046  and the sync project wiki page https://github.com/internetarchive/openlibrary/wiki/archive.org-%E2%86%94-Open-Library-synchronisation

This code is possibly only useful as a historical record now that the sync processes are better.

Making the PR as the code only existed on my personal fork.

### CODE REVIEW

**Comment by hornc** [comment]
What this code does not do yet is fill in duplicate archive.org scans with only `openlibrary` if the OLID is already linked to a different scan.

A final pass will be needed to fill in that edition and work data on the archive.org side.

File: ia-sync-bot/update-ocaid.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+           continue
+       if end and count > end:
+           break
+       # check and add ocaid to OL edition
+       print("Adding %s to %s" % (ocaid, olid))
+       edition = ol.get(olid)
+       #while not hasattr(edition, 'title')
+       try:
+           assert edition.title
+       except:
```
Comment (cclauss) [inline ia-sync-bot/update-ocaid.py:44]: Bare exception.

Comment (hornc) [reply ia-sync-bot/update-ocaid.py]: Thanks, fixed!

**Review by cclauss — APPROVED** [review]
LGTM

File: ia-sync-bot/requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+isbnlib
```
Comment (cclauss) [inline ia-sync-bot/requirements.txt:1]:
```suggestion
isbnlib
openlibrary-client
```

File: ia-sync-bot/.gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+ol_dump_editions_latest.*
+*.lst
+*.txt
```
Comment (cclauss) [inline ia-sync-bot/.gitignore:3]: Does this mean that git will ignore ia-sync-bot/requirements.txt?

**Pull request merged**